when dispatcher config input VATZ must not have panic when config input isn't appropriate

### Checklist
- [x] Bug from Service (Vatz)
- [ ] Bug from Plugin (Vatz) 
- [ ] Malfunction on CI/CD 
- [ ] etc(e.g, Hardware version doesn't match, )

---

### **Describe the bug**
> A clear and concise description of what the bug is.


### **How to Reproduce**
> Steps to reproduce the behavior:

1. Make build
2. leave a blank for dispatchers default.yaml 
3. start VATZ
4. Error occurs

```
 ./vatz start                   
2023-09-29T23:16:08-05:00 INF Initialize Server module=main
2023-09-29T23:16:08-05:00 INF Start VATZ Server on Listening Port: :9090 module=main
2023-09-29T23:16:08-05:00 INF start metric server: 127.0.0.1:18080 module=main
2023-09-29T23:16:08-05:00 INF start rpc server module=rpc
2023-09-29T23:16:08-05:00 INF start gRPC gateway server 127.0.0.1:19091 module=rpc
2023-09-29T23:16:08-05:00 INF start gRPC server 127.0.0.1:19090 module=rpc
2023-09-29T23:16:12-05:00 ERR Channel(Discord): Connection failed due to Invalid discord webhook address module=dispatcher
2023-09-29T23:16:13-05:00 ERR Channel(Pagerduty): Connection failed due to Error: HTTP response with status code: 400, JSON unmarshal object body failed: invalid character 'I' looking for beginning of value, body: Invalid routing key module=dispatcher
panic: interface conversion: interface {} is nil, not bool

goroutine 23 [running]:
github.com/dsrvlabs/vatz/manager/dispatcher.(*telegram).SendNotification(0xc000276380, {{0x1009641a4, 0xa}, 0x4, {0x100967dc7, 0x10}, 0x3, {0xc000256e70, 0xb}, 0xc00038e3f0})
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/dispatcher/telegram.go:121 +0x91f
github.com/dsrvlabs/vatz/manager/healthcheck.(*healthChecker).PluginHealthCheck(0x100f45b80, {0x100a75ba8, 0xc000136008}, {0x100a72458, 0xc00033d1a0}, {{0xc000256e70, 0xb}, {0xc000256e57, 0x9}, 0x4, ...}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/manager/healthcheck/general_healthcheck.go:69 +0x5b8
github.com/dsrvlabs/vatz/cmd.multiPluginExecutor({{0xc000256e70, 0xb}, {0xc000256e57, 0x9}, 0x4, 0x5, 0x2329, {0xc00033c6f0, 0x1, 0x1}}, ...)
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:146 +0x4d9
created by github.com/dsrvlabs/vatz/cmd.startExecutor
        /Users/dongyookang/ffplay/GolandProjects/xellos/vatz/cmd/start.go:125 +0x87
 ✘ dongyookang DK 👑   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-312-re ±
 

```


### **Expected behavior**
> A clear and concise description of what you expected to happen.

Throw error is okay, but not panic. 

### **Screenshots**
> If applicable, add screenshots to help explain your problem.

<img width="1356" alt="image" src="https://github.com/dsrvlabs/vatz/assets/6308023/d93b971c-7b4f-42bb-9cf9-e8cccebeae7c">
<img width="1912" alt="image" src="https://github.com/dsrvlabs/vatz/assets/6308023/d9b3bb19-ba6d-4faa-8c67-c57bff2f0f68">

--- a/cmd/start.go
+++ b/cmd/start.go
@@ -7,6 +7,7 @@ import (
 	"net"
 	"os"
 	"strconv"
+	"strings"
 	"time"
 
 	managerPb "github.com/dsrvlabs/vatz-proto/manager/v1"
@@ -142,7 +143,13 @@ func multiPluginExecutor(plugin config.Plugin, singleClient pluginPb.PluginClien
 		select {
 		case <-verifyTicker.C:
 			if pluginState.IsEnabled {
-				live, _ := healthChecker.PluginHealthCheck(ctx, singleClient, plugin, dispatchers)
+				live, err := healthChecker.PluginHealthCheck(ctx, singleClient, plugin, dispatchers)
+				if err != nil {
+					if strings.Contains(err.Error(), "Failed to send all configured notifications") {
+						executeTicker.Stop()
+						os.Exit(1)
+					}
+				}
 				if live == tp.AliveStatusUp {
 					okToSend = true
 				} else {

--- a/manager/dispatcher/discord.go
+++ b/manager/dispatcher/discord.go
@@ -3,6 +3,7 @@ package dispatcher
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
 	"strings"
@@ -127,10 +128,13 @@ func (d *discord) SendNotification(msg tp.ReqMsg) error {
 		_, err = c.Do(req)
 		if err != nil {
 			log.Error().Str("module", "dispatcher").Msgf("Channel(Discord): Send notification error: %s", err)
+			errorMessage := fmt.Sprintf("Channel(Discord) Error: %s", err)
+			return errors.New(errorMessage)
 		}
 		// TODO: Should handle response status.
 	} else {
 		log.Error().Str("module", "dispatcher").Msg("Channel(Discord): Connection failed due to Invalid discord webhook address")
+		return errors.New("Channel(Discord) Error: Invalid discord webhook address. ")
 	}
 	return nil
 }

--- a/manager/dispatcher/telegram.go
+++ b/manager/dispatcher/telegram.go
@@ -3,7 +3,6 @@ package dispatcher
 import (
 	"bytes"
 	"encoding/json"
-	"errors"
 	"fmt"
 	"io"
 	"net/http"
@@ -116,8 +115,7 @@ Plugin Name: <em>%s</em>
 
 	if response.StatusCode < 200 || response.StatusCode > 299 {
 		log.Error().Str("module", "dispatcher").Msgf("Channel(Telegram): Error in Response with Error code: %d", response.StatusCode)
-		errorMessage := fmt.Sprintf("REST API Error with HTTP response status code: %d", response.StatusCode)
-		return errors.New(errorMessage)
+		return fmt.Errorf("REST API Error with HTTP response status code: %d", response.StatusCode)
 	}
 
 	body, err = io.ReadAll(response.Body)

--- a/manager/dispatcher/telegram.go
+++ b/manager/dispatcher/telegram.go
@@ -124,14 +124,16 @@ Plugin Name: <em>%s</em>
 		return err
 	}
 	respJSON := make(map[string]interface{})
-	if !respJSON["ok"].(bool) {
-		log.Error().Str("module", "dispatcher").Msg("Channel(Telegram): Connection failed due to Invalid telegram token.")
-	}
 	err = json.Unmarshal(body, &respJSON)
 	if err != nil {
 		log.Error().Str("module", "dispatcher").Msgf("Channel(Telegram): Unmarshalling JSON Error: %s", err)
 		return err
 	}
+	if !respJSON["ok"].(bool) {
+		log.Error().Str("module", "dispatcher").Msg("Channel(Telegram): Connection failed due to Invalid telegram token.")
+		return fmt.Errorf("Invalid telegram token. ")
+	}
+
 	return nil
 }
 

--- a/manager/dispatcher/telegram.go
+++ b/manager/dispatcher/telegram.go
@@ -3,6 +3,7 @@ package dispatcher
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"net/http"
@@ -112,6 +113,13 @@ Plugin Name: <em>%s</em>
 		return err
 	}
 	defer response.Body.Close()
+
+	if response.StatusCode < 200 || response.StatusCode > 299 {
+		log.Error().Str("module", "dispatcher").Msgf("Channel(Telegram): Error in Response with Error code: %d", response.StatusCode)
+		errorMessage := fmt.Sprintf("REST API Error with HTTP response status code: %d", response.StatusCode)
+		return errors.New(errorMessage)
+	}
+
 	body, err = io.ReadAll(response.Body)
 	if err != nil {
 		log.Error().Str("module", "dispatcher").Msgf("Channel(Telegram): body parsing Error: %s", err)

--- a/manager/healthcheck/general_healthcheck.go
+++ b/manager/healthcheck/general_healthcheck.go
@@ -2,18 +2,17 @@ package healthcheck
 
 import (
 	"context"
-	"github.com/rs/zerolog/log"
-	"os"
-	"sync"
-	"time"
-
+	"errors"
 	pb "github.com/dsrvlabs/vatz-proto/plugin/v1"
 	"github.com/dsrvlabs/vatz/manager/config"
 	dp "github.com/dsrvlabs/vatz/manager/dispatcher"
 	tp "github.com/dsrvlabs/vatz/manager/types"
 	"github.com/dsrvlabs/vatz/utils"
 	"github.com/robfig/cron/v3"
-	emptypb "google.golang.org/protobuf/types/known/emptypb"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"sync"
+	"time"
 )
 
 var (
@@ -76,8 +75,9 @@ func (h *healthChecker) PluginHealthCheck(ctx context.Context, gClient pb.Plugin
 		}
 
 		if len(dispatchers) == errorCount {
+			errorMSG := "Failed to send all configured notifications."
 			log.Error().Str("module", "healthcheck").Msg("All Dispatchers failed to send a notifications, Please, Check your dispatcher configs.")
-			os.Exit(1)
+			return isAlive, errors.New(errorMSG)
 		}
 	}
 

--- a/manager/healthcheck/general_healthcheck.go
+++ b/manager/healthcheck/general_healthcheck.go
@@ -2,6 +2,8 @@ package healthcheck
 
 import (
 	"context"
+	"github.com/rs/zerolog/log"
+	"os"
 	"sync"
 	"time"
 
@@ -65,8 +67,17 @@ func (h *healthChecker) PluginHealthCheck(ctx context.Context, gClient pb.Plugin
 	}
 
 	if sendMSG {
+		errorCount := 0
 		for _, dispatcher := range dispatchers {
-			dispatcher.SendNotification(deliverMSG)
+			sendNotificationError := dispatcher.SendNotification(deliverMSG)
+			if sendNotificationError != nil {
+				errorCount = errorCount + 1
+			}
+		}
+
+		if len(dispatchers) == errorCount {
+			log.Error().Str("module", "healthcheck").Msg("All Dispatchers failed to send a notifications, Please, Check your dispatcher configs.")
+			os.Exit(1)
 		}
 	}
 

--- a/manager/healthcheck/healthcheck_test.go
+++ b/manager/healthcheck/healthcheck_test.go
@@ -93,14 +93,18 @@ func TestPluginHealthCheckFailed(t *testing.T) {
 		status, err := h.PluginHealthCheck(ctx, &mockPluginCli, config.Plugin{Name: "test"}, mockDispatchers)
 
 		// Asserts
-		assert.Nil(t, err)
+		assert.Error(t, err)
 		assert.Equal(t, tp.AliveStatusDown, status)
 
 		statuses := h.PluginStatus(ctx)
 
-		assert.Equal(t, 1, len(statuses))
-		assert.Equal(t, "test", statuses[0].Plugin.Name)
-		assert.Equal(t, tp.AliveStatusDown, statuses[0].IsAlive)
+		assert.Equal(t, 0, len(statuses))
+
+		/*
+			Both cases are not available since there's error when no appropriate configs for notification
+			assert.Equal(t, "test", statuses[0].Plugin.Name)
+			assert.Equal(t, tp.AliveStatusDown, statuses[0].IsAlive)
+		*/
 
 		mockPluginCli.AssertExpectations(t)
 	}

